netbox_inventory/forms/filters.py

```diff
@@ -103,10 +103,19 @@ class AssetFilterForm(NetBoxModelFilterSetForm):
         null_option='None',
         label='Tenant',
     )
-    contact_id = DynamicModelMultipleChoiceField(
+    contact_group = DynamicModelMultipleChoiceField(
+        queryset=ContactGroup.objects.all(),
+        required=False,
+        null_option='None',
+        label='Contact Group',
+    )
+    contact = DynamicModelMultipleChoiceField(
```

Comment:
If you select just a contact group and don't select any contacts, netbox will try to filter asset list by contact group by generating URL: `/plugins/inventory/assets/?contact_group=1`. Because AssetFilterSet does not support this filter it will just return all assets.

I suggest you:

- rename this field to contact_group_id
- add `contact_group_id` `ModelMultipleChoiceFilter` to `AssetFilterSet` in `filtersets.py` that will filter on `contact__group`. See filters for asset supplier for example how to do this.